### PR TITLE
landlock countries fix

### PR DESCRIPTION
**Pull request by DeniseGiub**
## Checklist

- [x] I tested my contribution locally and it seems to work fine.
- [x] Code and workflow changes are sufficiently documented.
- [x] Newly introduced dependencies are added to `envs/environment.yaml` and `envs/environment.docs.yaml`.
- [x] A note for the release notes `doc/release_notes.rst` is amended in the format of previous release notes, including reference to the requested PR.

### CODE REVIEW

File: scripts/build_bus_regions.py [hunk 1]
```diff
@@ -210,87 +140,50 @@ def get_id(coords):
...
 
         c_b = n.buses.country == country
-        if n.buses.loc[c_b & n.buses.substation_lv, ["x", "y"]].empty:
-            _logger.warning(f"No low voltage buses found for {country}!")
-            continue
 
         onshore_shape = country_shapes[country]
         onshore_locs = n.buses.loc[c_b & n.buses.substation_lv, ["x", "y"]]
-        if snakemake.config["cluster_options"]["alternative_clustering"]:
-            onshore_geometry = get_gadm_shape(onshore_locs, gadm_shapes)[0]
```
Comment (pz-max) [inline scripts/build_bus_regions.py]: This deletes the GADM clustering functionalities (see our preprint)
We could create a test for checking `["alternative_clustering"]: True` to detect such breaking changes in future.

**Review by pz-max — CHANGES_REQUESTED** [review]
Changes look good. However, the code would break if the config.yaml clusters GADM shapes. To do: 
- [x] create a test for alternative_clustering: True
- [ ] Make the whole workflow working with alternative_clusterin: True (undo a couple of changes)

**Comment by davide-f** [comment]
> Changes look good. However, the code would break if the config.yaml clusters GADM shapes. To do:
> 
> * [ ]  create a test for alternative_clustering: True
> * [ ]  Make the whole workflow working with alternative_clusterin: True (undo a couple of changes)

@DeniseGiub Great Job!

@pz-max for flexibility purposes, rather than creating a new file, which may lead to duplication, we could do some hacking to create it dynamically with the CI.
The other test config would be copy and paste of the other one, with few parameters changed; if we change the config file then we have to change 2 files instead of one.
This is more "hacking", we can also start with a duplicate of the test, that should be easy for Denise to do.
@DeniseGiub to do that, you would need to add a test config file in the test folder and modify the .github/workflow/* config files to run also the additional script, see https://github.com/pypsa-meets-earth/pypsa-earth/blob/e35545c3bcfa14b6eef714f28d95ef807db450cb/.github/workflows/ci-linux.yaml#L64-L68

**Comment by pz-max** [comment]
> @pz-max for flexibility purposes, rather than creating a new file, which may lead to duplication, we could do some hacking to create it dynamically with the CI. The other test config would be copy and paste of the other one, with few parameters changed; if we change the config file then we have to change 2 files instead of one.

We redesigned the testing procedure in this PR: https://github.com/pypsa-meets-earth/pypsa-earth/pull/466
It will be merged after a successful CI run.

**Pull request closed without merging**